### PR TITLE
Data Cache

### PR DESCRIPTION
**Pull request by JeanPierreNell**
Added Data Cache for Issue : #76 ([Performance testing - embedding graph object identifiers in the payload](https://github.com/frmscoe/General-Issues/issues/76))

Fixed Issue: #110 ([Fix account_holder edge creation out of the data preparation service](https://github.com/frmscoe/General-Issues/issues/110))

### CODE REVIEW

File: src/logic.service.ts [hunk 1]
```diff
@@ -57,9 +59,17 @@ export const handlePain001 = async (transaction: Pain001): Promise<any> => {
...
-      cacheDatabaseClient.addAccountHolder(creditorId, creditorAcctId, CreDtTm),
-      cacheDatabaseClient.addAccountHolder(debtorId, debtorAcctId, CreDtTm),
+      cacheDatabaseClient.addAccountHolder(creditorId, creditorHash, CreDtTm),
+      cacheDatabaseClient.addAccountHolder(debtorId, debtorHash, CreDtTm)
     ]);
+
+    transaction.DataCache = {
+      "cdtrId": creditorId, 
+      "dbtrId": debtorId,
+      "cdtrAcctId": creditorAcctId,
```
Comment (johanfol) [inline src/logic.service.ts]: here you set cdtrAcctId to the actual value, instead of Hash

File: src/logic.service.ts [hunk 2]
```diff
@@ -117,6 +127,11 @@ export const handlePain013 = async (transaction: Pain013): Promise<any> => {
     throw err;
   }
 
+  transaction.DataCache = {
+    "cdtrAcctId": creditorHash,
```
Comment (johanfol) [inline src/logic.service.ts]: Here you set cdtrAcctId to the hash, instead of the actual id

File: src/logic.service.ts [hunk 3]
```diff
@@ -170,6 +185,11 @@ export const handlePacs008 = async (transaction: Pacs008): Promise<any> => {
     throw err;
   }
 
+  transaction.DataCache = {
+    "dbtrAcctId": debtorHash,
+    "cdtrAcctId": creditorHash
```
Comment (johanfol) [inline src/logic.service.ts]: Here you set cdtrAcctId to the hash, instead of the actual id

File: src/logic.service.ts [hunk 4]
```diff
@@ -219,6 +239,11 @@ export const handlePacs002 = async (transaction: Pacs002): Promise<any> => {
     throw err;
   }
 
+  transaction.DataCache = {
+    "cdtrAcctId": transactionRelationship.from,
```
Comment (johanfol) [inline src/logic.service.ts]: Here you set cdtrAcctId to the accounts/hash , instead of the actual id or hash.

**Review by johanfol — CHANGES_REQUESTED** [review]
It's a bit strange how the same field is being populated with different values throughout the message flow - from Pain 001 to Pacs 002 the fields are constantly populdated differently. Let's chat.

File: src/logic.service.ts (hunk 4 above)
Comment (JeanPierreNell) [reply src/logic.service.ts]: This is now retrieved from Redis/Database

File: src/logic.service.ts (hunk 2 above)
Comment (JeanPierreNell) [reply src/logic.service.ts]: This is now retrieved from Redis/Database

File: src/logic.service.ts (hunk 3 above)
Comment (JeanPierreNell) [reply src/logic.service.ts]: This is now retrieved from Redis/Database

**Review by johanfol — APPROVED** [review]

**Pull request merged**